### PR TITLE
Adds support for WDS Require Login plugin, if available

### PR DESCRIPTION
**Pull request by phatsk**
Allows two options for the Require Login key/token to be used when creating a request.

### CODE REVIEW

**Comment by jtsternberg** [comment]
This doesn't need the class_exists check because the request side does not need that plugin.

**Comment by jtsternberg** [comment]
Also it looks like you're mixing object properties with `$args` array parameters.

**Comment by phatsk** [comment]
@jtsternberg 
> This doesn't need the class_exists check because the request side does not need that plugin.

Was just keeping the object clutter down, but I can remove the check

> Also it looks like you're mixing object properties with $args array parameters.

Not sure about that part but...should be out of the way now either way.

**Comment by jtsternberg** [comment]
Line 537 is using object properties when the 'require_login_key'  and 'require_login_token' are set as values of the arms property. Also, I think we ought to change to a more generic 'header_key' and 'header_token' as they don't have to be specific to the require-login plugin.

File: wds-wp-json-api-connect.php [hunk 1]
```diff
@@ -527,7 +529,15 @@ public function cache_api_description_for_json_url() {
 				return $this->json_desc;
 			}
 
-			$this->response = wp_remote_get( $this->args['json_url'] );
+			$get_args = array(
+				'headers' => array(),
+			);
+
+			if ( strlen( $this->require_login_key ) && strlen( $this->require_login_token ) ) {
+				$get_args['headers'][ $this->require_login_key ] = $this->require_login_token;
```
Comment (phatsk) [inline wds-wp-json-api-connect.php]: > Line 537 is using object properties when the 'require_login_key' and 'require_login_token' are set as values of the arms property

Ah I see - they are also set as accessible via the magic __get on 897 and 898. Blinded me to the use of $this->args right above. Will change that.

**Comment by jtsternberg** [comment]
Modified to make this super generic (allow passing any headers) and closed in b5270fca650af2c1c0efd5926666f1d21d4f56b4.

**Pull request closed without merging**